tests/go-validation/v1apitests/provisioning_test.go

```diff
@@ -0,0 +1,444 @@
+	s.clusterNames = append(s.clusterNames, clusterName)
+
+	provisioningClient, err := clients.NewProvisioningClient(s.bearerToken)
+	require.NoError(s.T(), err)
+
+	clusterConfig := cluster.NewRKE2ClusterConfig(clusterName, s.clusterNamespace, cluster.CNI, cloudCredentialName, cluster.KubernetesVersion, s.rkeMachinePools)
+
+	clusterObj := cluster.NewCluster(s.clusterNamespace, provisioningClient)
+	s.clusterObjs = append(s.clusterObjs, clusterObj)
+
+	s.T().Logf("Creating Cluster %s", clusterName)
+	v1Cluster, err := clusterObj.CreateCluster(clusterConfig)
+	require.NoError(s.T(), err)
+
+	assert.Equal(s.T(), v1Cluster.Name, clusterName)
+
+	s.T().Logf("Created Cluster %s", v1Cluster.Name)
+
+	getCluster, err := clusterObj.PollCluster(v1Cluster.Name)
+	require.NoError(s.T(), err)
+
+	s.v1Cluster = getCluster
+}
+
+type ProvisioningTestSuite struct {
+	suite.Suite
+
+	// server config
+	host       string
+	password   string
+	adminToken string
+	userToken  string
+	cni        string
+	k8sVersion string
+
+	// Digital Ocean variables
+	digitalOceanCloudCredential       *cloudcredentials.CloudCredential
+	digitalOceanCloudCredentialConfig *corevV1.Secret
+
+	// Node Driver variables
+	rkeMachinePools []apisV1.RKEMachinePool
+	machineConfig   *unstructured.Unstructured
+
+	// AWS environment variables
+	awsInstanceType    string
+	awsRegion          string
+	awsRegionAZ        string
+	awsAMI             string
+	awsSecurityGroup   string
+	awsSSHKeyName      string
+	awsCICDInstanceTag string
+	awsIAMProfile      string
+	awsUser            string
+	awsVolumeSize      int
+
+	ec2Client        *aws.EC2Client
+	nodes            []*aws.EC2Node
+	nodeConfig       *cluster.NodeConfig
+	nodeRoles        []string
+	baseEC2Name      string
+	baseClusterName  string
+	clusterNamespace string
+	randString       string
+
+	// rest api variables
+	bearerName       string
+	bearerToken      string
+	managementClient *v3.Client
+
+	clusterNames []string
+	clusterObjs  []*cluster.Cluster
+	v1Cluster    *apisV1.Cluster
+
+	rancherCleanup bool
+}
+
+// The SetupSuite method will be run by testify once, at the very
+// start of the testing suite, before any tests are run.
+func (s *ProvisioningTestSuite) SetupSuite() {
+	s.host = clients.Host
+	s.password = user.Password
+	s.adminToken = clients.AdminToken
+	s.userToken = clients.UserToken
+	s.cni = cluster.CNI
+	s.k8sVersion = cluster.KubernetesVersion
+	s.clusterNamespace = "fleet-default"
+	s.rancherCleanup = environmentvariables.RancherCleanup()
+
+	testifyArg := os.Args[5:len(os.Args)]
+
+	var testCase string
+	if testifyArg[0] == "-testify.m" {
+		testCase = testifyArg[1]
+	} else {
+		testCase = ""
+	}
+
+	if strings.Contains(testCase, "DigitalOcean") || testCase == "" {
+		for userName, bearerToken := range clients.BearerTokensList() {
+
+			s.T().Logf("Create Cloud Credential for user %s", userName)
+			client, err := clients.NewCoreV1Client(bearerToken)
+			require.NoError(s.T(), err)
```

Comment:
We may need to opt for an alternative to using `require` during testify's hook functions (setup, teardown, before and after). See: https://github.com/stretchr/testify/issues/849 for details.

Reply:
I'll look into this once the cloud credential stuff is working

Reply:
How it works now, if setup fails the test just fails out. I think that's fine, it's going to try to run the test with a failed setup.